i6_models/parts/frontend/generic_frontend.py

```diff
@@ -0,0 +1,224 @@
+
+    def __post__init__(self):
+        super().__post_init__()
+        self.check_valid()
+
+
+class GenericFrontendV1(nn.Module):
+    def __init__(self, model_cfg: GenericFrontendV1Config):
+        """
+        Generic Front-End
+        can be used to generate customized frontend by combining convolutional and pooling layers, as well as activation
+        functions differently
+
+        To get the ESPnet case, for example Conv2dSubsampling6, use these options
+            layer_ordering = [FrontendLayerType.Conv2d, FrontendLayerType.Conv2d]
+            conv_kernel_sizes = [3, 5]
+            strides = [2, 3]
+
+        To get the i6_models VGG4LayerActFrontendV1, use the options:
+            layer_ordering = [FrontendLayerType.Conv2d, FrontendLayerType.Activation, FrontendLayerType.Pool2d,
+                FrontendLayerType.Conv2d, FrontendLayerType.Conv2d, FrontendLayerType.Activation, FrontendLayerType.Pool2d]
+            conv_kernel_sizes = [3, 3, 3]
+            conv_out_dims = [32, 34, 64]
+            pool_kernel_sizes = [3, 3]
+            pool_strides = [2, 2]
+            activations = [torch.nn.ReLU(), torch.nn.ReLU()]
+        """
+        super().__init__()
+
+        model_cfg.check_valid()
+
+        self.cfg = model_cfg
+
+        self.frontend_layers = nn.ModuleList([])
+
+        conv_layer_index = 0
+        pool_layer_index = 0
+        activation_layer_index = 0
+        last_channel_dim = 1
+        last_feat_dim = model_cfg.in_features
+        for layer_type in model_cfg.layer_ordering:
+            if layer_type == FrontendLayerType.Conv2d:
+                conv_out_dim = model_cfg.conv_out_dims[conv_layer_index]
+                conv_kernel_size = model_cfg.conv_kernel_sizes[conv_layer_index]
+                conv_stride = 1 if model_cfg.conv_strides is None else model_cfg.conv_strides[conv_layer_index]
+                conv_padding = (
+                    get_same_padding(conv_kernel_size)
+                    if model_cfg.conv_paddings is None
+                    else model_cfg.conv_paddings[conv_layer_index]
+                )
+
+                self.frontend_layers.append(
+                    nn.Conv2d(
+                        in_channels=last_channel_dim,
+                        out_channels=conv_out_dim,
+                        kernel_size=conv_kernel_size,
+                        stride=conv_stride,
+                        padding=conv_padding,
+                    )
+                )
+
+                last_channel_dim = conv_out_dim
+                last_feat_dim = calculate_output_dim(
+                    in_dim=last_feat_dim,
+                    filter_size=conv_kernel_size if isinstance(conv_kernel_size, int) else conv_kernel_size[1],
+                    stride=conv_stride if isinstance(conv_stride, int) else conv_stride[1],
+                    padding=conv_padding if isinstance(conv_padding, int) else conv_padding[1],
```

Comment:
```suggestion
                    filter_size=conv_kernel_size[1],
                    stride=conv_stride[1],
                    padding=conv_padding[1],
```
If you decide to adopt the "tuple-only-parameter" suggestion.